### PR TITLE
Refactore avr reconciler

### PR DESCRIPTION
**Pull request by BenamarMk**
- Added an AVR instance per AVR reciliation
- Added lastKnownState to the AVR.Status

### CODE REVIEW

File: controllers/applicationvolumereplication_controller.go [hunk 1]
```diff
@@ -485,15 +502,14 @@ func (r *ApplicationVolumeReplicationReconciler) findManifestWork(
 	return nil, nil
 }
 
-func (r *ApplicationVolumeReplicationReconciler) deleteExistingManfiestWork(
-	avr *ramendrv1alpha1.ApplicationVolumeReplication,
+func (a *AVRInstance) deleteExistingManfiestWork(
```
Comment (raghavendrabhat) [inline controllers/applicationvolumereplication_controller.go:505]: (nit) Can you please change the name of this function from "deleteExistingManfiestWork" to "deleteExistingManifestWork" ? Its a small typo in the name, but helps in searching.

Comment (BenamarMk) [reply controllers/applicationvolumereplication_controller.go:505]: good catch.  Thanks

**Review by ShyamsundarR — APPROVED** [review]
As I see it the state change in status is not used anywhere and stored for future use. So as this is scaffolding to build the next layer, LGTM!

**Review by raghavendrabhat — APPROVED** [review]
LGTM

**Pull request merged**